sycl/source/detail/scheduler/scheduler.hpp

```diff
@@ -19,6 +19,152 @@
 #include <set>
 #include <vector>
 
+/// \defgroup sycl_graph DPC++ Execution Graph
+///
+/// DPC++, unlike OpenCL, provides a programming model in which the user doesn't
+/// need to manage dependencies between kernels and memory explicitly. The DPC++
+/// Runtime must ensure correct execution with respect to the order commands are
+/// submitted.
+///
+/// This document describes the part of the DPC++ Runtime that is responsible
+/// for building and processing dependency graph.
+///
+/// ## A couple of words about DPC++ and SYCL execution and memory model
+///
+/// The SYCL framework defines command group (\ref CG) as an entity that
+/// represents minimal execution block. The command group is submitted to SYCL
+/// queue and consists of a kernel or an explicit memory operation, and their
+/// requirements. The SYCL queue defines the device and context using which the
+/// kernel should be executed.
+///
+/// The commands that contain explicit memory operations include copy, fill,
+/// update_host and other operations. It's up to implementation how to define
+/// these operations.
+///
+/// The relative order of command groups submission defines the order in which
+/// kernels must be executed if their memory requirements intersect. For
+/// example, if a command group A writes to a buffer X, command group B reads
+/// from X, then the scheduled execution order of A and B will be the same as
+/// their dynamic submission order (matches program order if submitted from the
+/// same host thread).
+///
+/// Memory requirements are requests to SYCL memory objects, such as buffer and
+/// image. SYCL memory objects are not bound to any specific context or device,
+/// it's SYCL responsibility to allocate and/or copy memory to the target
+/// context to achieve correct execution.
+///
+/// Refer to SYCL Specification 1.2.1 sections 3.4 and 3.5 to find more
+/// information about SYCL execution and memory model.
+///
+/// ### Example of DPC++ application
+///
+/// \code{.cpp}
+/// {
+///   // Creating DPC++ CPU and GPU queues
```

Comment:
BTW, I agree with @keryell here: if some thing comes from the SYCL spec and it wasn't changed by our extensions or something, then we need to refer to it as `SYCL`, not `DPC++`.

See how it was done during renaming ([this comment](https://github.com/intel/llvm/pull/1249#issuecomment-596897155)):

> I've tried to be careful doing all replacements in this change and keep SYCL everywhere when referring to SYCL standard and SYCL language concepts.